lib/data_cite/metadata.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module DataCite
+  class Metadata
+    class Error < StandardError; end
+    class ValidationError < Error; end
+
+    attr_reader :work_version
+
+    RESOURCE_TYPES = {
+      Work::Types::DATASET => 'Dataset'
+    }.freeze
+
+    def initialize(work_version:)
+      @work_version = work_version
+      @work = work_version.work
+    end
+
+    def attributes
+      {
+        titles: [{ title: work_version.title }],
+        creators: [creator],
+        publicationYear: publication_year,
+        types: {
+          resourceTypeGeneral: resource_type
+        }
+      }
+    end
+
+    def validate!
+      raise ValidationError.new("title can't be blank") if attributes[:titles].map { |t| t[:title] }.all?(&:blank?)
+
+      raise ValidationError.new("publicationYear can't be blank") if attributes[:publicationYear].blank?
+
+      raise ValidationError.new("Unknown mapping for work type: #{work.work_type.inspect}") if attributes.dig(
+        :types, :resourceTypeGeneral
+      ).blank?
+    end
+
+    def valid?
+      validate! && true
+    rescue Error
+      false
+    end
+
+    private
+
+      attr_reader :work
+
+      def creator
+        {
+          givenName: work.depositor.given_name,
+          familyName: work.depositor.surname
+        }
+      end
+
+      def publication_year
+        date = parsed_publication_date || work_version.created_at
+        date&.year
+      end
+
+      def parsed_publication_date
+        Time.zone.parse(work_version.published_date.first)
```

Comment:
@rschenk Any reason why we wouldn't use DateTime?

Reply:
`Time.zone` is how you hook into Rail's timezone support, `DateTime` doesn't have an equivalent feature. Also since Ruby doesn't actually have a time-of-day class (as you'd assume Time would be), the difference between the two is fuzzy at best